docs/interfaces/auth-service.txt

```diff
@@ -61,44 +95,44 @@ Authenticate (token: Token, cookie: Cookie): {userId: number; sessionId: string;
 *
 * @throws InvalidCredentials
 */
-WhoAmI(cookie: Cookie): Token;
+api/authenticate (Ticket): Response<LoginInformation>;
 
 /**
-* Function to kill one specific session by its id.
-* 
-* An exception is thrown, if the client has not the necessary permissions to make this action.
-* Also, if there is no session with the given id, an exception is thrown.
-*
-* @throws NoPermissions: Only users themselves can clear their own session and (super-) admins
-* can do this, too.
+* Function to sign out one specific client from a user by its corresponding session-id.
 */
-ClearSessionById (sessionId: string, cookie: Cookie, token: Token): void publishes LogoutSessionEvent;
+api/clear-session-by-id (sessionId: string, Ticket): Response<void> publishes LogoutSessionEvent;
 
 /**
-* Function to kill all current opened sessions except the one, which is requesting.
-*
-* An exception is thrown, if the client has not the necessary permissions to make this action.
-* 
-* @throws NoPermissions: Only (super-) admins has the necessary permissions to logout and clear
-* other user's session.
+* Function to kill all current opened sessions from one user except the one, which is requesting.
 */ 
-ClearAllSessionsExceptThemselves (token: Token, cookie: Cookie): void publishes LogoutSessionEvent;
-
-Event LogoutSessionEvent on topic Logout {
-    sessionId: string;
-}
+api/clear-all-sessions-except-themselves (sessionId: string, Ticket): Response<void> publishes LogoutSessionEvent;
 
 /**
 * The service deletes the session depending on the given Token.
 *
 * @throws InvalidCredentials
 */
-Logout (token: Token, cookie: Cookie): void publishes LogoutSessionEvent;
+api/logout (Ticket): Response<void> publishes LogoutSessionEvent;
 
 /**
 * Returns all currently active sessions.
 *
 * @throws NoPermissions: The users can only see their own session. Only (super-) admins can see sessions
 * of other users.
 */
-ListSessions (token: Token, cookie: Cookie): string[];
\ No newline at end of file
+api/list-sessions (Ticket): Response<string[]>;
+
+/**
+* Hashes a given value with an optional salt. A random salt (64bit) is generated, if a salt is not given.
+*
+* @returns the hashed value. The hashed value is structured as follows: [salt + hash].
+*/
+hash (toHash: string, salt?: string): Response<{hash: string}>;
```

Comment:
Use `oldHash` or so instead of salt. The user of you service does not know how you hash is composed

Reply:
In combination with `is-quals`, you do not need a salt here at all. This would than always give a random salt. For new passwords, this is ok.

Reply:
Why should I need an `oldHash` here? Another service can easily get a hashed value. There is `is-equals` to check, if the hash of a given value  is equals to a stored hash.

Reply:
Sorry, the reference is outdated. When I wrote the comment, the interface was different.